### ISSUE TITLE
Use ABI filters for armv7 + x86 builds (was: ~4x APK size increase from including :service-fxa in v4.1-LAT1+)

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/firefox-tv#2647]
### Vision statement / What / Requirements
According to GH releases, the v4.0 APK was 5.43 MB and the v4.1-LAT1 APK was 21.1MB. The release notes say the only thing that changed was an a-c upgrade, though other things could have landed on master that weren't taken into account. Unfortunately, the changelog hasn't been updated for v4.1 so we can't see what's in the entire build: #2646.

I suspect we landed FxA sign in, which includes some heavy rust components. However, 15MB seems like a lot for that.

### Impact
We could be unnecessarily shipping more than we need to. a-c might be accidentally shipping more than they need to.

Bigger APKs generally means fewer installs from users because they're waiting longer for the download, more hard drive space taken up for them, and (negligibly? I'm not sure) worse performance as the app uses more RAM & resources.

### Acceptance criteria
- Use ABI filters to make an armv7 (device) & x86 build (emulator): this is expected to reduce the APK size to ~10 MB.

**Comment by mcomella** [comment]
I took a look with APK analyzer and found four `lib/*/libfxaclient_ffi.so` files in the APK totaling 15.3MB together: as suspected, it was probably the introduction of FxA that caused this. afaik, we only support one architecture so if we can remove the other architectures, that'd be a big improvement. Then again, ~3.5+ MB for the FxA lib also seems like a lot to me so we should ask about that too.

I think the next steps are on the FxA team: I'll file a bug.

**Comment by mcomella** [comment]
I filed https://github.com/mozilla/fxa/issues/2115

**Comment by mcomella** [comment]
This issue is now being tracked upstream at https://github.com/mozilla/application-services/issues/1559.

In that issue, we received an update: they'll try slimming down the builds but it's challenging for them to remove architectures: we'll have to do that part ourselves. More details to come in that issue.

**Comment by mcomella** [comment]
Partially unblocked: we can try to strip the unused architectures from our build. We're awaiting details in the other bug but could be more aggressive about getting them or figuring it out ourselves.

@athomasmoz Which column should we move this to?

**Comment by mcomella** [comment]
To be explicit, this build with the APK size increase is currently being shipped to users.

**Comment by athomasmoz** [comment]
@mcomella I saw a comment in the other ticket. Is this unblocked now? If that's the case, I'd like to put it into our current sprint, even if that means taking another ticket out

**Comment by mcomella** [comment]
@athomasmoz Yes, this is unblocked.

**Comment by mcomella** [comment]
Here's the comment that says what we can do about the .so files we don't care about: https://github.com/mozilla/application-services/issues/1560#issuecomment-520117001

**Comment by mcomella** [comment]
Summary what I know so far:
- We only ship to arm v7
- We need x86 builds to run on the emulator
- Users may side load other configurations (do we want to support these?)
- Native code is used for `libjnidispatch.so` (I'm not sure what this is: maybe all JNI apps ship with it?) and FxA.

By default, Android builds a Fat APK: it has all the architectures.

We have two options:
- ABI filters, which whitelists architectures for the Fat APK
- Split APK, which builds a separate APK for each architecture

I think we should do the latter to create the slimest APKs but it creates complexities:
- Developers need to build x86 for emulator and armv7 for devices
- Release owners need to upload arm v7 builds to Amazon dashboard & ?all builds to GitHub releases
- Users sideloading need to know their architecture (unless we also post fat APKs)

**Comment by mcomella** [comment]
Official ABI filter/APK split docs are available here: https://developer.android.com/ndk/guides/abis

**Comment by mcomella** [comment]
We decided to reduce the amount of work necessary for this issue: use ABI filters and we'll file a follow-up for APK splits.
